source/integrate_with_govuk_pay/index.html.md.erb

```diff
@@ -0,0 +1,170 @@
+* payment journeys your users have started, and links to services they have
+requested
+* the status of payment journeys, in particular when journeys are completed
+and the outcome
+
+### Service backend
+
+The service backend is server-side software which should:
+
+* make API calls to GOV.UK Pay to start users’ payment journeys
+* store information about user payment journeys in the datastore
+* redirect the user to the `next_url` provided by GOV.UK Pay, where the user
+will enter their payment information and confirm their payment
+* receive users’ requests when they are redirected back to your service via
+the `return_url`
+* identify the returning user via the session
+* make a call to the GOV.UK Pay API to determine the outcome of the payment
+* display any information about the outcome of the payment and next steps to
+the user
+
+### Finance and accounting systems
+
+You may wish to integrate your finance and accounting systems with GOV.UK Pay
+using the API. For example you could:
+
+* automatically fetch data about the outcome of payment journeys, and import
+that into your finance system so that payments can be reconciled against bank
+transaction information
+* connect a customer-relationship management (CRM) or case-management system
+to GOV.UK Pay, so that staff can issue refunds from within your system.
+
+You may want to read more in the sections on [__Reporting__](/reporting) and
+[__Refunding payments__](/refunding_payments).
+
+### APIs (GOV.UK Pay)
+
+The public GOV.UK Pay API offers a set of simple operations to conduct and
+report on payments. You can read about the API in the [__API
+reference__](/api_reference) section of this documentation and use the
+interactive [API explorer](https://gds-payments.gelato.io/api-explorer/)
+[external link]. You can also read more about [__Reporting__](/reporting).
+
+## Mop-up jobs
+
+A mop-up job is a background process which checks the outcome of incomplete
+payment journeys.
+
+Most services need to release their service to the user automatically, upon
+successful completion of a payment. If you do this then you should plan for
+inevitable failures associated with unreliable internet connections and user
+behaviour. For example:
+
+* user abandons their payment journey before completing it
+* user completes their payment successfully, but their network connection is
+interrupted before they return to your service
+
+In these cases, users will never visit the `return_url` on your service. You
+should therefore not rely on the `return_url` to determine the outcome of the
+payment journey.
+
+### Using a mop-up job
+
+This section describes the recommended approach for most integrations. It
+provides the best guarantee that edge cases are handled properly and
+automatically.
+
+In order to use a mop-up job, you should have a datastore which keeps track of
+incomplete payment journeys.
+
+The following diagram gives a detailed overview of the sequence:
```

Comment:
could be clearer that the following diagram shows the complete payment journey flow

Reply:
the final line of the diagram says 'display confirmation screen (success/fail)' - i'm perhaps getting into pedantry here: in payment failure cases, gov.uk pay will have already shown an appropriate payment unsuccessful screen, so we'd expect the service not to show another fail screen, but to direct the user to the next sensible point in the journey (e.g. retrying the payment).

Reply:
also, there's a line in the sequence diagram "find incomplete journey using journey_id" - is that in the right place? i thought you wouldn't see any incomplete journeys in this flow, because the user doesn't get redirected on incomplete journeys..?

Reply:
1. 'The following diagram gives a detailed overview of the payment flow including use of a datastore' 
2. 'display appropriate screen' _might_ be better 
3. ignore my comment